avocado/core/task/statemachine.py

```diff
@@ -214,7 +183,7 @@ async def check_finished_dependencies(runtime_task):
 
             # dependencies finished, let's check if they finished
             # successfully, so we can move on with the parent task
-            dependencies_ok = await check_finished_dependencies(runtime_task)
+            dependencies_ok = runtime_task.is_possible_to_run()
```

Comment:
Just a matter of taste, so feel free to ignore, but to make this method shorter, I would go with "can_run()".